pkg/client/config/config.go

```diff
@@ -123,9 +123,16 @@ func loadConfig(context string) (*rest.Config, error) {
 	// try the in-cluster config.
 	kubeconfigPath := os.Getenv(clientcmd.RecommendedConfigPathEnvVar)
 	if len(kubeconfigPath) == 0 {
-		if c, err := loadInClusterConfig(); err == nil {
+		c, err := loadInClusterConfig()
+		if err == nil {
 			return c, nil
 		}
+
+		defer func() {
+			if configErr != nil {
+				log.Error(err, "unable to load in-cluster config")
```

Comment:
@alvaroaleman 
Q: Does this really work as expected?

I think it could potentially pick up errors produced later in the func in l. 150 or l.155 

Wondering if we could simply use `if err != nil` to log this error instead